src/libpmemfile-core/os_thread.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2016-2017, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * os_locks.h -- wrappers around system locking functions
+ */
+
+#ifndef PMEMFILE_OS_LOCKS_H
+#define PMEMFILE_OS_LOCKS_H 1
+
+typedef union {
+	long long align;
+	char padding[64];
+} os_mutex_t;
+
+typedef union {
+	long long align;
+	char padding[64];
```

Comment:
Are these used internally only?
And in volatile memory?
I suspect yes.
Then please, don't create such a placeholder type, no need for specifying 64 byte size.
Please just typedef the mutex available on the platform, as in:
https://github.com/tinycthread/tinycthread/blob/master/source/tinycthread.h#L194

The 64 byte assumption is just plain wrong, there is no standard anywhere, that says this thing should fit in 64 bytes. There is no such magic number.

E.g. it might as well be 200 bytes:
https://github.com/pmem/issues/issues/427